src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.WebSearch/Main.cs

```diff
@@ -0,0 +1,412 @@
+                    || input.All(char.IsDigit))
+                {
+                    return false;
+                }
+
+                try
+                {
+                    _ = new UriBuilder(input);
+                }
+                catch (UriFormatException)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+        }
+
+        private bool IsDefaultBrowserSet()
+        {
+            return !string.IsNullOrEmpty(_browserPath);
+        }
+
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+            UpdateBrowserIconPath(_context.API.GetCurrentTheme());
+        }
+
+        public string GetTranslatedPluginTitle()
+        {
+            return Properties.Resources.plugin_name;
+        }
+
+        public string GetTranslatedPluginDescription()
+        {
+            return Properties.Resources.plugin_description;
+        }
+
+        private void OnThemeChanged(Theme oldtheme, Theme newTheme)
+        {
+            UpdateIconPath(newTheme);
+            UpdateBrowserIconPath(newTheme);
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "Design",
+            "CA1031:Do not catch general exception types",
+            Justification = "We want to keep the process alive but will log the exception")]
+        private void UpdateBrowserIconPath(Theme newTheme)
+        {
+            try
+            {
+                string progId = GetRegistryValue(
+                    "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\http\\UserChoice",
+                    "ProgId");
+
+                if (progId.StartsWith("Opera", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Opera user preferences file:
+                    string prefFile;
+
+                    if (progId.Contains("GX", StringComparison.OrdinalIgnoreCase))
+                    {
+                        _browserName = "Opera GX";
+                        prefFile = System.IO.File.ReadAllText($"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\\Opera Software\\Opera GX Stable\\Preferences");
+                    }
+                    else
+                    {
+                        _browserName = "Opera";
+                        prefFile = System.IO.File.ReadAllText($"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\\Opera Software\\Opera Stable\\Preferences");
+                    }
+
+                    string url = ExtraStringMethods.GetJSONStringValue(prefFile, new string[] { "default_search_provider_data", "template_url_data", "url" });
+
+                    if (url is null)
```

Comment:
When do we use this search url? I thought we have the special argument `?` for starting the browsers in search mode. 🤔

Reply:
This for Opera. The argument `?` doesn't work for it.